Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/triangulate_faces.h

```diff
@@ -219,8 +219,26 @@ class Triangulate_polygon_mesh_modifier
 
       const FT p1p3 = cross_product(p2-p1, p3-p2) * cross_product(p0-p3, p1-p0);
       const FT p0p2 = cross_product(p1-p0, p1-p2) * cross_product(p3-p2, p3-p0);
-      halfedge_descriptor res = (p0p2>p1p3) ?  CGAL::Euler::split_face(v0, v2, pmesh)
-                                            :  CGAL::Euler::split_face(v1, v3, pmesh);
+
+      halfedge_descriptor res = boost::graph_traits<PolygonMesh>::null_halfedge();
+
+      if (p1p3!=p0p2)
```

Comment:
Maybe we should have a predicate. Because here we compare computed values, without an exact kernel.

Reply:
How do you want to call it?    `Comparison_result  fct(const Point_& p0,..... p3)`

Reply:
Also the code might profit from computing some differences only once.

Reply:
I don't think there is any need to use an exact predicate here. I don't think we could select an invalid diagonal.